doc/how_to/custom_components/reactive_esm/reactive_esm_layout.md

```diff
@@ -0,0 +1,41 @@
+# Create Layouts With ReactiveESM
+
+In this guide we will show you how to build custom layouts using HTML and `ReactiveESM`.
+
+## Layout a single parameter
+
+You can layout a single object as follows.
+
+```{pyodide}
+import panel as pn
```

Comment:
How would I create layouts using Panel components as parameter values? When I attempt I get `can't serialize <class 'panel.widgets.indicators.Dial'`.

Reply:
Will push a how-to in a bit but right now it's basically:

- Declare a parameter as `param.ClassSelector(class_=Viewable)`

Also considering adding an actual parameter for Panel components and for lists of Panel components.

Reply:
Looking to this. Currently I cannot figure out how to do this.

Reply:
I'm still wrestling with this a little bit, it should be easy to add a list of objects but that's not yet possible, here's the minimum child rendering example:

```python
import param
import panel as pn

from panel import ReactiveESM

class Example(ReactiveESM):

    child = param.ClassSelector(class_=pn.viewable.Viewable)

    _esm = """
    export function render({ children }) {
      const div = document.createElement("button");
      div.appendChild(children.child.el)
      return div
    }"""

Example(child=pn.panel('A Markdown pane!')).servable()
```

Reply:
And the equivalent `React` based version:

```
import param
import panel as pn

from panel import ReactiveESM

class Example(ReactiveESM):

    child = param.ClassSelector(class_=pn.viewable.Viewable)

    _esm = """
    function App(props) {
      return (
        <button>{props.child}</button>
      )
    }

    export function render({ children }) {
      return <App child={children.child}></App>
    }"""

Example(child=pn.panel('A Markdown pane!')).servable()
```

Reply:
Also `children` should probably just contain the DOM nodes rather than the bokeh views.

Reply:
```python

class JSExample(ReactiveESM):

    objects = param.List(item_type=pn.viewable.Viewable)

    _esm = """
    export function render({ children }) {
      const div = document.createElement('div')
      div.append(...children.objects)
      return div
    }"""

class ReactExample(ReactiveESM):

    objects = param.List(item_type=pn.viewable.Viewable)

    _esm = """
    export function render({ children }) {
      return <div>{children.objects}</div>
    }"""

```

Reply:
Just a thought. Would it be possible to `merge` data and `children`? Why have two? It makes it harder to explain the api.

Reply:
How do I `.watch` for changes? Children does not seem to have a `.watch` method.

Reply:
How would I make a `grid` like layout like `GridSpec`?

Reply:
It seems I cannot narrow the parameter type to a list of `Viewable` classes.

```python
import param
import panel as pn

from panel import ReactiveESM

class Example(ReactiveESM):

    child = param.ClassSelector(class_=(pn.pane.Markdown, pn.pane.HTML))

    _esm = """
    export function render({ children }) {
      const button = document.createElement("button");
      button.appendChild(children.child)
      return button
    }"""

Example(child=pn.panel('A Markdown pane!')).servable()
```

![image](https://github.com/holoviz/panel/assets/42288570/a8b3e854-dc68-45c1-85fd-ba03bf3d00f0)

Reply:
Same problem for `item_type`.

```python
import param

import panel as pn

from panel import ReactiveESM


class Example(ReactiveESM):

    objects = param.List(item_type=(pn.pane.Markdown, pn.widgets.Button))

    _esm = """
    export function render({ children }) {
      const div = document.createElement('div')
      div.append(...children.objects)
      return div
    }"""


Example(
    objects=[pn.panel("A **Markdown** pane!"), pn.widgets.Button(name="Click me!")]
).servable()
```

![image](https://github.com/holoviz/panel/assets/42288570/23a7afc6-1745-4d8e-947b-f853fae72929)